engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/TypeDefinitionSet.scala

```diff
@@ -0,0 +1,54 @@
+package pl.touk.nussknacker.engine
+
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.{Invalid, Valid}
+import org.springframework.expression.spel.SpelNode
+import pl.touk.nussknacker.engine.api.expression.ExpressionParseError
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+import pl.touk.nussknacker.engine.definition.{DefinitionExtractor, ProcessDefinitionExtractor, TypeInfos}
+import pl.touk.nussknacker.engine.spel.TypedNode
+import pl.touk.nussknacker.engine.spel.ast.SpelAst.RichSpelNode
+
+
+class TypeDefinitionSet(typeDefinitions: Map[String, TypeInfos.ClazzDefinition]) {
+
+  def validateTypeReference(spelNode: SpelNode): Validated[NonEmptyList[ExpressionParseError], SpelNode] = {
```

Comment:
let's have TypeReference as parameter to this method

Reply:
Changed it, thanks